trade-finance-manager-ui/server/controllers/case/cancellation/check-details.controller.ts

```diff
@@ -76,11 +71,15 @@ export const postDealCancellationDetails = async (req: PostDealCancellationDetai
       return res.redirect('/not-found');
     }
 
-    if (!canSubmissionTypeBeCancelled(deal.dealSnapshot.submissionType)) {
+    const submissionTypeCanBeCancelled = canSubmissionTypeBeCancelled(deal.dealSnapshot.submissionType);
+    const cancellationIncomplete = reason === undefined || !bankRequestDate || !effectiveFrom;
```

Comment:
```suggestion
    const cancellationIncomplete = !reason || !bankRequestDate || !effectiveFrom;
```

Reply:
`reason` can be an empty string

Reply:
`!reason` should work for undefined and empty string?

Reply:
Exactly, I want to rule out `undefined` but allow an empty string. Using your suggestion if the user doesnt provide a reason, it will be set to `''` and then they wont be able to submit the cancellation

Reply:
Ah, sorry is "reason" an optional field?

Reply:
Yep its optional, so will always have a value of at least an empty string

Reply:
Thanks @MarRobSoftwire , didn't realise, makes complete sense 😄